### PR TITLE
Migrate TravisCI to Github Actions

### PR DESCRIPTION
**Pull request by Griffk97**

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        npm run compile
+    - name: Lint & Typecheck
+      run: npm run check 
+    - name: Unit Test
+      run: npm run test:coverage
+    - name: Coverage
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+      env:
```
Comment (Drew-Kimberly) [inline .github/workflows/node.js.yml:34]: Add a `publish` step that is conditional on the `push` event happening on the `master` branch. Just echo "hello world" for now.

Conditional steps: https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        npm install
+        npm run compile
+    - name: Lint & Typecheck
+      run: npm run check 
+    - name: Unit Test
+      run: npm run test:coverage
+    - name: Coverage
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Drew-Kimberly) [inline .github/workflows/node.js.yml:33]: question: where is this secret coming from? I don't see any configured in the repo settings

**Review by Drew-Kimberly — CHANGES_REQUESTED** [review]
Nice work so far!

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      env:
+        CI: true
```
[the remark this answers is not in the document]
Comment (Griffk97) [reply .github/workflows/node.js.yml:35]: fixed

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (Griffk97) [reply .github/workflows/node.js.yml:33]: GitHub actions uses a per job automatically generated token ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow https://github.com/coverallsapp/github-action

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      uses: actions/cache@v2
+      with:
+        path: |
+          node_modules
+          packages/**/node_modules
+        key: node_modules-${{ hashFiles('**/package-lock.json') }}
+
+    - name: Compile
+      run: |
+        npm install
```
[the remark this answers is not in the document]
Comment (Griffk97) [reply .github/workflows/node.js.yml]: fixed

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        node-version: 12.x
+    - name: restore lerna
+      uses: actions/cache@v2
+      with:
+        path: |
+          node_modules
+          packages/**/node_modules
+        key: node_modules-${{ hashFiles('**/package-lock.json') }}
+
+    - name: Compile
```
[the remark this answers is not in the document]
Comment (Griffk97) [reply .github/workflows/node.js.yml:28]: fixed

File: .github/workflows/node.js.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      run: |
+        npm install
+        npm run compile
+    - name: Lint & Typecheck
```
[the remark this answers is not in the document]
Comment (Griffk97) [reply .github/workflows/node.js.yml]: fixed

[the remark this answers is not in the document]
Comment (Griffk97) [reply .github/workflows/node.js.yml]: fixed

File: .github/workflows/node.js.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+    - name: restore lerna
+      uses: actions/cache@v2
+      with:
+        path: |
+          node_modules
+          packages/**/node_modules
+        key: node_modules-${{ hashFiles('**/package-lock.json') }}
```
[the remark this answers is not in the document]
Comment (Griffk97) [reply .github/workflows/node.js.yml:23]: fixed

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+jobs:
+  compile-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+    - name: restore lerna
```
[the remark this answers is not in the document]
Comment (Griffk97) [reply .github/workflows/node.js.yml]: fixed

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (Griffk97) [reply .github/workflows/node.js.yml]: added

**Review by Drew-Kimberly — APPROVED** [review]

**Comment by Drew-Kimberly** [comment]
[Migrate TravisCI to Github Actions](https://trello.com/c/nnEVVuzP/2-migrate-travisci-to-github-actions)

**Pull request merged**